src/main/scala/com/github/sbt/sbom/licenses/LicensesArchive.scala

```diff
@@ -1,23 +1,36 @@
 package com.github.sbt.sbom.licenses
 
+import com.github.sbt.sbom.licenses.LicensesArchive.normalizeUrl
+
 import scala.io.Source
 
 class LicensesArchive(licenses: Seq[License]) {
-  private val licensesByUrl: Map[String, License] = licenses.foldLeft(Map[String, License]()) { (map, license) =>
-    map ++ license.references.foldLeft(Map[String, License]()) { (map, ref) =>
-      map + (ref -> license)
-    }
-  }
+  private val licensesByNormalizedUrl: Map[String, License] =
+    licenses.iterator.flatMap { license =>
+      license.references.map { reference =>
+        (normalizeUrl(reference), license)
+      }
+    }.toMap
 
-  def findByUrl(url: String): Option[License] = licensesByUrl.get(url)
+  def findByNormalizedUrl(url: String): Option[License] = licensesByNormalizedUrl.get(normalizeUrl(url))
```

Comment:
Do we really need this method? It seems only to be used in tests, or am I misunderstanding that?